pkg/test/zq_example.go

```diff
@@ -0,0 +1,224 @@
+			// else.
+		}
+		return ast.WalkContinue, nil
+	})
+
+	if command != nil && err == nil {
+		err = fmt.Errorf("%s without a following %s", ZQ_COMMAND, ZQ_OUTPUT)
+	}
+	return examples, err
+}
+
+// Return the text of a ast.FencedCodeBlock as a string.
+func BlockString(fcb *ast.FencedCodeBlock, source []byte) string {
+	var b strings.Builder
+	for i := 0; i < fcb.Lines().Len(); i++ {
+		line := fcb.Lines().At(i)
+		b.Write(line.Value(source))
+	}
+	return b.String()
+}
+
+// Translate a zq-command example to a runnable command including abspath to zq
+// binary and globs turned into absolute file paths.
+func QualifyCommand(command string) ([]string, error) {
+	command = strings.TrimSpace(command)
+	command = strings.Split(command, "#")[0]
+
+	var delim string
+	doublequoteindex := strings.Index(command, `"`)
+	if doublequoteindex == -1 {
+		doublequoteindex = len(command)
+	}
+	singlequoteindex := strings.Index(command, `'`)
+	if singlequoteindex == -1 {
+		singlequoteindex = len(command)
+	}
+	if singlequoteindex < doublequoteindex {
+		delim = `'`
+	} else {
+		delim = `"`
+	}
+
+	pieces := strings.Split(command, delim)
+	if len(pieces) != 3 {
+		return nil, fmt.Errorf("could not split zq command 3 tokens: %s", command)
+	}
+
+	command_and_flags := strings.Split(strings.TrimSpace(pieces[0]), " ")
+	if command_and_flags[0] != "zq" {
+		return nil, fmt.Errorf("command does not start with zq: %s", command)
+	}
+	// Nice, but this makes unit testing more complicated.
+	zq, err := ZqAbsPath()
+	if err != nil {
+		return nil, err
+	}
+	command_and_flags[0] = zq
+
+	zql := strings.TrimSpace(pieces[1])
+
+	var fileargs []string
+	sampledata, err := ZqSampleDataAbsPath()
+	if err != nil {
+		return nil, err
+	}
+	for _, relglobarg := range strings.Split(strings.TrimSpace(pieces[2]), " ") {
+		files, err := filepath.Glob(filepath.Join(sampledata, "zeek-default", relglobarg))
+		if err != nil {
+			return nil, err
+		}
+		fileargs = append(fileargs, files...)
+	}
+
+	finalized := command_and_flags
+	finalized = append(finalized, zql)
+	finalized = append(finalized, fileargs...)
+	return finalized, nil
+}
+
+// Generate ZQ example test cases from ZQ example pairs found in a file.
+func TestcasesFromFile(filename string) ([]ZQExampleTest, error) {
+	var tests []ZQExampleTest
+	var examples []ZQExamplePair
+	filename, err := filepath.Abs(filename)
+	if err != nil {
+		return nil, err
+	}
+	repopath, err := RepoAbsPath()
+	if err != nil {
+		return nil, err
+	}
+	source, err := ioutil.ReadFile(filename)
+	if err != nil {
+		return nil, err
+	}
+	reader := text.NewReader(source)
+	parser := goldmark.DefaultParser()
+	doc := parser.Parse(reader)
+	examples, err = CollectExamples(doc, source)
+	if err != nil {
+		return nil, err
+	}
+	for i, example := range examples {
+		name := strings.TrimPrefix(filename, repopath)
+		name = strings.TrimPrefix(name, string(filepath.Separator)) + strconv.Itoa(i+1)
```

Comment:
I don't understand how this prefix (separator + index) would be present on the filename?

Reply:
Done at d3dd5c7 .